src/core/mac/mac.cpp

```diff
@@ -107,19 +107,14 @@ otError ChannelMask::GetNextChannel(uint8_t &aChannel) const
     return error;
 }
 
-const char *ChannelMask::ToString(char *aBuffer, uint16_t aSize) const
+ChannelMask::InfoString ChannelMask::ToString(void) const
```

Comment:
Perhaps keep original return result of `AsCString()` for all `ToString()` calls?
It seems almost all uses of `ToString()` are actually `.ToString().AsCString()`.
I do like how the new structure removes the need to pass a buffer and size.

Reply:
We need `String` object to be returned so that the object and its underlying buffer are in scope at the call side and the pointer to the buffer is valid.